src/Factory/RepositoryProvider.php

```diff
@@ -8,23 +8,42 @@
 use Cycle\ORM\ORMInterface;
 use Cycle\ORM\RepositoryInterface;
 use Cycle\ORM\SchemaInterface;
+use Psr\Container\ContainerInterface;
 use Yiisoft\Di\Container;
-use Yiisoft\Di\Support\ServiceProvider;
+use Yiisoft\Di\CompositeContainer;
+use Yiisoft\Di\Contracts\ServiceProviderInterface;
 
 use function is_string;
 
 /**
  * This provider provides factories to the container for creating Cycle entity repositories.
  * Repository list is compiled based on data from the database schema.
  */
-final class RepositoryProvider extends ServiceProvider
+final class RepositoryProvider implements ServiceProviderInterface
 {
-    public function register(Container $container): void
+    public function getDefinitions(): array
     {
-        /** @var ORMInterface */
-        $orm = $container->get(ORMInterface::class);
-        /** @psalm-suppress InaccessibleMethod */
-        $container->setMultiple($this->getRepositoryFactories($orm));
+        return [];
+    }
+
+    /**
+     * @return array<Closure>
+     */
+    public function getExtensions(): array
+    {
+        return [
+            ContainerInterface::class => function (ContainerInterface $container, ContainerInterface $extended) {
+                /** @var ORMInterface */
+                $orm = $extended->get(ORMInterface::class);
+                /** @psalm-suppress InaccessibleMethod */
+                $repositoryContainer = new Container($this->getRepositoryFactories($orm));
+                $compositeContainer = new CompositeContainer();
+                $compositeContainer->attach($repositoryContainer);
```

Comment:
What's the benefit of using composite container here?

Reply:
What's another way to add repositories definitions to the container?

Reply:
Ah, right. We don't have `set` as public :)

Reply:
That should be documented in DI readme.

Reply:
Can we not wrap everything in a composite container in the modules themselves?
It would be better to provide the container in this package, and the application itself would add it to the main composite container

Reply:
Important but postponed it for now.

Reply:
Will check it later. Not within these PRs.

Reply:
I don't like this PR but you can feel fre to merge it.